src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/ILanguageServer.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using StreamJsonRpc;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public interface ILanguageServer : IAsyncDisposable
+{
+    event EventHandler<bool>? Shutdown;
+
+    event EventHandler? Exit;
+
+    InitializeParams ClientSettings { get; }
+
+    /// <summary>
+    /// Handle the LSP initialize request by storing the client capabilities and responding with the server
+    /// capabilities.  The specification assures that the initialize request is sent only once.
+    /// </summary>
+    [JsonRpcMethod(Methods.InitializeName, UseSingleObjectParameterDeserialization = true)]
+    Task<InitializeResult> InitializeAsync(InitializeParams initializeParams, CancellationToken cancellationToken);
```

Comment:
Hmmm, if we were to get rid of the `InitializeAsync` bit here and the `ClientSettings` above would we be able to decouple this assembly from the VS.Protocol assembly? Or would that even be worth it? Just a stray thought in my head (don't even know if it's a problem) is the version of this DLL will be directly tied to the LSP protocol version. Meaning, if LSP protocol revs you'd need to update thie CLASP assembly too

Reply:
Hmm, that's an interesting idea I'll have to play around with. I feel like including `InitializeAsync` on the ILanguageServer itself should maybe be avoided if possible anyway.